tests/integration/test_main.py

```diff
@@ -77,3 +77,22 @@ def test_microcluster_leader_down(juju: jubilant.Juju):
         juju.remove_unit("microovn/1")
     juju.add_unit("microovn")
     juju.wait(jubilant.all_active)
+
+def test_certificates_integration(juju: jubilant.Juju):
+    juju.deploy(microovn_charm_path)
+    juju.add_unit("microovn")
+    juju.deploy(token_distributor_charm_path)
+    juju.deploy("self-signed-certificates")
+    juju.integrate("microovn","microcluster-token-distributor")
+    juju.integrate("microovn","self-signed-certificates")
+    juju.wait(jubilant.all_active)
+    juju.wait(lambda _: "Pushed certificate to workload" in juju.debug_log())
+    destination = juju.status().apps["microovn"].units["microovn/1"].public_address
+    destination = destination + ":6643"
+    try:
+        juju.exec("openssl s_client -connect {0} -verify_return_error".format(destination), unit="self-signed-certificates/0")
+        assert False, "Expected to fail, somehow passed"
+    except:
+        pass
+    output = juju.exec("openssl s_client -connect {0} -CAfile /tmp/ca-cert.pem -verify_return_error".format(destination), unit="self-signed-certificates/0")
```

Comment:
I wonder if there is a separate argument needed for openssl `s_client` / `verify` commands for them to verify certificate chains with intermediate certificates not in the global trust store.

Whether this is a feature of those commands or if it would have implications for the target client implementation too, i'm not sure. So we might benefit from a one-off check and comment with the result. Neutron uses ovsdbapp which uses the openvswitch Python client implementation.